kvmagent/kvmagent/plugins/nvram/nvram.py

```
@@ -74,6 +84,53 @@ def cleanup(self):
+        result.type = to.type
+
+        if not nvram_common.check_nvram_vm_host_file_path_format(result.path):
+            result.error = 'invalid nvram vm host file path ' + result.path
+            return result
+        vm_host_file.read_vm_host_file_base64(result)
+        return result
+    
+    def write_file(self, to):
+        # type: (vm_host_file.VmHostFileTO) -> None
+        if not nvram_common.check_nvram_vm_host_file_path_format(to.path):
+            raise Exception('invalid nvram vm host file path ' + to.path)
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)